doc/pages/user/041-astarte_device_sdks.md

```diff
@@ -0,0 +1,590 @@
+    AstarteDevice device =
+        new AstarteGenericDevice(
+            deviceId,
+            realm,
+            credentialsSecret,
+            new ExampleInterfaceProvider(),
+            pairingUrl,
+            connectionSource);
+
+    // ExampleMessageListener listens for device connection, disconnection and failure.
+    device.setAstarteMessageListener(new ExampleMessageListener());
+
+    // Connect the device
+    device.connect();
+
+```
+
+Python:
+```python
+# declare device options
+device = Device(device_id, realm, credentials_secret, pairing_base_url)
+
+# load device interfaces
+device.add_interface(json.loads("/examples/interfaces/com.example.Interface.json"))
+
+#register a callback that will be invoked everytime the device successfully connects
+device.on_connected(callback)
+
+#connect the device asynchronously
+device.connect()
+```
+
+Rust:
+```rust
+    /// declare device options
+    let mut sdk_options =
+        AstarteOptions::new(&realm, &device_id, &credentials_secret, &pairing_url);
+
+    /// load interfaces from a directory
+    sdk_options
+        .add_interface_files("./examples/interfaces")
+        .unwrap();
+
+    /// build Astarte client
+    sdk_options.build().await.unwrap();
+
+    /// connect the device
+    let mut device = sdk_options.connect().await.unwrap();
+```
+
+## Streaming data
+
+All Astarte Device SDKs have a primitive for sending data to a remote Astarte instance.
+
+### Using Individual Aggregated Interfaces
+In Astarte interfaces with `individual` aggregation, each mapping is treated as an independent value 
+and is managed individually.
+
+Following examples show how to send a value that will be inserted into the `"/test0/value"` time series
+which is defined by `"/%{sensor_id}/value"` parametric endpoint (that is part of
+`"org.astarte-platform.genericsensors.Values"` datastream interface).
+
+C (ESP32):
+```c
+struct timeval tv;
+gettimeofday(&tv, NULL);
+uint64_t ts = tv->tv_sec * 1000 + tv->tv_usec / 1000;
+
+astarte_err_t err = astarte_device_stream_double_with_timestamp(device, "org.astarte-platform.genericsensors.Values", "/test0/value", 0.3, ts, 0);
+```
+
+C++ (Qt5):
+```c++
+m_sdk->sendData("org.astarte-platform.genericsensors.Values", "/test0/value", 0.3, QDateTime::currentDateTime());
+```
+
+Elixir:
+```elixir
+Device.send_datastream(pid, "org.astarte-platform.genericsensors.Values", "/test0/value", 0.3, timestamp: DateTime.utc_now())
+```
+
+Go:
+```go
+d.SendIndividualMessageWithTimestamp("org.astarte-platform.genericsensors.Values", "/test0/value", 0.3, time.Now())
+```
+
+Java:
+```java
+genericSensorsValuesInterface.streamData("/test0/value", 0.3, DateTime.now());
+```
+
+Python:
+```python
+device.send("org.astarte-platform.genericsensors.Values", "/test0/value", 0.3, timestamp=datetime.now())
+```
+
+Rust
+```rust
+device.send_with_timestamp("org.astarte-platform.genericsensors.Values", "/test0/value", 0.3, chrono::offset::Utc::now())
```

Comment:
`...).await?;`

Reply:
Also, sending float is not obvious since it can fail, you can change `0.3` with `AstarteType::Double(0.3)` or just send an integer since it's an example